wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheConnection.java

```diff
@@ -134,6 +148,14 @@ public CacheConnection(final Properties properties) {
     this.cacheIamRegion = CACHE_IAM_REGION.getString(properties);
     this.cacheUsername = CACHE_USERNAME.getString(properties);
     this.cachePassword = CACHE_PASSWORD.getString(properties);
+    this.cacheConnectionTimeout = Duration.ofMillis(CACHE_CONNECTION_TIMEOUT.getInteger(properties));
+    this.cacheConnectionPoolSize = CACHE_CONNECTION_POOL_SIZE.getInteger(properties);
+    if(this.cacheConnectionPoolSize <= 0) {
```

Comment:
I would prefer if you throw an IllegalArgumentException if the user tries to set the pool size to be > MAX with an appropriate error message.